shared/torngit/github.py

```diff
@@ -518,16 +527,71 @@ def _parse_response(self, res: Response):
                 )
                 return res.text
 
-    def _get_next_fallback_token(self, fallback_idx: int) -> Optional[str]:
+    def _possibly_mark_current_installation_as_rate_limited(
+        self,
+        *,
+        reset_timestamp: Optional[str] = None,
+        retry_in_seconds: Optional[int] = None,
+    ) -> None:
+        current_token = self.token
+        if current_token.get("installation_id") is not None:
+            installation_id = current_token["installation_id"]
+            if retry_in_seconds is None and reset_timestamp is None:
+                log.warning(
+                    "Can't mark installation as rate limited because TTL is missing",
+                    extra=dict(installation_id=installation_id),
+                )
+                return
+            ttl_seconds = retry_in_seconds
+            if ttl_seconds is None:
+                # https://docs.github.com/en/rest/using-the-rest-api/best-practices-for-using-the-rest-api?apiVersion=2022-11-28#handle-rate-limit-errors-appropriately
+                ttl_seconds = int(reset_timestamp) - int(
+                    datetime.datetime.now(datetime.timezone.utc).timestamp()
```

Comment:
[nit] this line is a bit hard to read, I think it might be easier to read if we imported what we were using from datetime only
```
from datetime import datetime, timezone
```